### ISSUE TITLE
Implemented interface responsible for name should be ordered first

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#323]
If  class implements multiple interfaces, and one interface is responsible to give the class its name, then that interface should be ordered first before all other interfaces

Example:
```c#
public class MySample : IDisposable, IMySample
{ }
```
should produce a violation.

To fix that, the code should be changed into:
```c#
public class MySample : IMySample, IDisposable
{ }
```

Note: This rule shall only apply if the class has the same name like the interface, except the `I` interface marker.
